### PR TITLE
Fix undefined reference to Kokkos::Profiling::profileLibraryLoaded()

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes the failures, e.g., in https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/Kokkos/detail/Kokkos/2023/pipeline.

### CODE REVIEW

**Comment by DavidPoliakoff** [comment]
(Obviously approved but wait for CI for merge)

**Comment by DavidPoliakoff** [comment]
It is weird though, how did the test make it through CI? This one, for example, passed: https://cloud.cees.ornl.gov/jenkins-ci/blue/organizations/jenkins/Kokkos/detail/Kokkos/2011/pipeline

File: core/src/impl/Kokkos_Profiling.cpp [hunk 1]
```diff
@@ -886,6 +886,11 @@ void declare_optimization_goal(const size_t context,
 #include <cstring>
 
 namespace Kokkos {
+
+namespace Profiling {
+bool profileLibraryLoaded() { return false; }
+}
```
Comment (dalg24) [inline core/src/impl/Kokkos_Profiling.cpp:892]:
```diff
diff --git a/core/src/impl/Kokkos_Profiling.cpp b/core/src/impl/Kokkos_Profiling.cpp
index 3e593c5..387b83e 100644
--- a/core/src/impl/Kokkos_Profiling.cpp
+++ b/core/src/impl/Kokkos_Profiling.cpp
@@ -889,7 +889,7 @@ namespace Kokkos {
 
 namespace Profiling {
 bool profileLibraryLoaded() { return false; }
-}
+}  // namespace Profiling
 
 namespace Tools {

Comment (masterleinad) [reply core/src/impl/Kokkos_Profiling.cpp]: Fixed.

**Pull request merged**